source/sass/patterns/organisms/content-header.scss

```diff
@@ -130,8 +130,8 @@
 .content-header__download_link {
   @include inset(0, inline-end);
   @include inset(24px, block);
-  position: absolute;
-  float: inline-end;
+  //position: absolute;
+  @include float(inline-end);
```

Comment:
@discodavey Removing the positioning, and floating instead, fixes the left/right problem. I've left in the `inset` values in case you need them for reference. You might need to find another way of tweaking the position of the icon.